start.py

```diff
@@ -0,0 +1,121 @@
+    # node disks map is missing any nodes.
+    if set(args.primary_node + args.secondary_nodes) != set(node_disks.keys()):
+        raise Exception('Not all nodes are accounted for in the --node-disks dictionary')
+
+    primary_node = Node(hostname=args.primary_node[0],
+                        group='primary',
+                        image=primary_node_image,
+                        ports=[{MCS_SERVER_PORT: MCS_SERVER_PORT}
+                               if args.predictable
+                               else MCS_SERVER_PORT],
+                        devices=node_disks.get(args.primary_node[0]))
+
+    secondary_nodes = [Node(hostname=hostname,
+                            group='secondary',
+                            image=secondary_node_image,
+                            devices=node_disks.get(hostname))
+                       for hostname in args.secondary_nodes]
+
+
+    cluster = Cluster(primary_node, *secondary_nodes)
+    cluster.primary_node = primary_node
+    cluster.start(args.network, pull_images=args.always_pull)
+
+    # Verify that service MapR warden is running and then only proceed.
+    logger.info('Check if Service MapR warden is running ...')
+    start_time = time()
+    stop_time = start_time + 30
+    while time() < stop_time:
+        exit_code = primary_node.execute('service mapr-warden status').exit_code
+        if exit_code == 0:
+            break
+        sleep(1)
+    
+    if exit_code != 0:
+        raise Exception('Service MapR warden is still not running. Exiting.')
+
+    add_mapr_user_command = ['useradd mapr',
+                             'echo mapr | passwd mapr --stdin',
+                             'cp -R /root/.ssh ~mapr',
+                             'chown -R mapr:mapr ~mapr/.ssh']
+    cluster.execute("bash -c '{}'".format('; '.join(add_mapr_user_command)))
+
+    logger.info('Stopping Warden and ZooKeeper on nodes ...')
+    cluster.execute('service mapr-warden stop')
+    primary_node.execute('service mapr-zookeeper stop')
+
+    logger.info('Generating new UUIDs ...')
+    cluster.execute('/opt/mapr/server/mruuidgen > /opt/mapr/hostid')
+
+    for node in cluster:
+        configure_command = ('/opt/mapr/server/configure.sh -C {0} -Z {0} -RM {0} -HS {0} '
+                             '-u mapr -g mapr -D {1}'.format(
+                                 primary_node.fqdn, 
+                                 ','.join(node_disks.get(node.hostname))
+                             ))
+        logger.info('Running %s on %s ...', configure_command, node.hostname)
+        node.execute("bash -c '{}'".format(configure_command))
+
+    logger.info('Waiting for MapR Control System server to come online ...')
+    def condition(address, port):
+            return socket().connect_ex((address, port)) == 0
+    def success(time):
+        logger.info('MapR Control System server is online in %s seconds.', time)
```

Comment:
=> `'MapR Control System server is online after %s seconds.'`

Reply:
Done.